### PR TITLE
[MIG][11.0] mail_digest

### PR DESCRIPTION
**Pull request by simahawk**
NOTE: I've tried to squash the 2 OCA transbot commits but I get tons of conflicts...

### CODE REVIEW

File: mail_digest/data/ir_cron.xml
Comment (grindtildeath) [inline mail_digest/data/ir_cron.xml]: <data> to remove.

File: mail_digest/models/mail_digest.py
Comment (grindtildeath) [inline mail_digest/models/mail_digest.py]: Looks like this can be removed, as it isn't used in the function.

File: mail_digest/models/mail_digest.py
Comment (grindtildeath) [inline mail_digest/models/mail_digest.py]: `mail_id` should be renamed `mail`

File: mail_digest/models/mail_digest.py
Comment (grindtildeath) [inline mail_digest/models/mail_digest.py]: `mail_id` is then `mail.id`

File: mail_digest/models/mail_digest.py
Comment (grindtildeath) [inline mail_digest/models/mail_digest.py]: Again `mail_id` > `mail`,  `message=None` to remove, as it's not used

File: mail_digest/models/mail_digest.py
Comment (grindtildeath) [inline mail_digest/models/mail_digest.py]: `mail_id=mail_id` > `mail=mail`

File: mail_digest/models/res_users.py
Comment (grindtildeath) [inline mail_digest/models/res_users.py]: Seems this should be removed.

**Review by grindtildeath — CHANGES_REQUESTED** [review]
It seems, there's something wrong with message and mail_id params, as a few are unused, Otherwise, looks good !

**Comment by simahawk** [comment]
@grindtildeath obsolete params removed. tnx :)

File: mail_digest/data/ir_cron.xml
Comment (grindtildeath) [inline mail_digest/data/ir_cron.xml]: Still this tag to remove.

File: mail_digest/data/config_param.xml
Comment (grindtildeath) [inline mail_digest/data/config_param.xml]: Still this tag to remove.

**Review by grindtildeath — COMMENTED** [review]
Two `<data>`XML tags to remove and it's fine for me :+1:

**Comment by simahawk** [comment]
@grindtildeath done

**Review by grindtildeath — APPROVED** [review]

**Review by mpanarin — APPROVED** [review]
LGTM

File: mail_digest/models/mail_digest.py
Comment (tschanzt) [inline mail_digest/models/mail_digest.py]: is there any reason to still use the partner here? specially since you then get the user_id from the partner.

Comment (simahawk) [reply mail_digest/models/mail_digest.py]: some reasons:
1. this method is called by res.partner `_notify` method as notification are sent to users but attached to partners. I like to keep the 2 things tied;
2. message/email recipients are partners (as in std notifications). Again like to keep this in line.

Comment (tschanzt) [reply mail_digest/models/mail_digest.py]: ok, that makes sense.

**Review by tschanzt — APPROVED** [review]

**Comment by hbrunn** [comment]
it's quite hard for me to separate the old commits from the new ones, @simahawk, can you check this?

**Comment by simahawk** [comment]
@hbrunn the last 3 commits. The former are improvements coming from v10 that were not ported yet.

**Comment by hbrunn** [comment]
thanks, I was doing something stupid before. I miss a migration script, but given we don't have rules about that to my knowledge, I won't block on this. Any reason not to provide one? Should be just a couple of SQL statements and make the upgrade experience much better.

**Comment by simahawk** [comment]
@hbrunn main reason, honestly: not needed for my project and not sure I can spend time on it.
I provided info in the README on how to achieve that... in case someone is willing to help. That could be even me in the next future :wink:

**Comment by hbrunn** [comment]
@simahawk fair enough. But you inspired me to make this PR: https://github.com/OCA/maintainer-tools/pull/324

**Pull request merged**
